lib/app.rb

```diff
@@ -161,4 +168,46 @@ def add_book
     puts 'Book created successfully!'
     @books << book
   end
+
+  def list_musicalbum
+    if @musicalbums.length.zero?
+      puts 'No Music Album added yet !'
+    else
+
+      @musicalbums.each_with_index do |album, index|
+        puts "#{index + 1} Music Album :"
+        puts album
+      end
+    end
+  end
+
+  def list_genre
+    if @genres.length.zero?
+      puts 'No Genre registered yet!'
+    else
+
+      @genres.each_with_index do |genre, index|
+        puts "#{index + 1}) Genre : #{genre}"
+      end
+    end
+  end
+
+  def create_musicalbum
+    genre, author, label = retrieve_objects
+    published_date = [(print 'Published date (yyyy-mm-dd): '), gets.rstrip][1]
+    year, month, day = published_date.split('-')
+    on_spotify = [(print 'Is this album on spotify (Type True or False): '), gets.rstrip][1]
+    begin
+      musicalbum = MusicAlbum.new(Date.new(year.to_i, month.to_i, day.to_i), on_spotify)
+      author.add_item(musicalbum)
+      @authors << author unless @authors.include?(author)
+      musicalbum.genre = genre
```

Comment:
Please check that the implementation of this is similar to the one for authors (the two lines above this one). Since at the moment you are not storing the genre in the `@genres` variable.

Here, the genre should be an object that you create in the method `retrieve_objects` from line 101. You can implement it similar to how authors are created.